azure-iot-device/azure/iot/device/iothub/models/twin.py

```diff
@@ -0,0 +1,28 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""This module contains classes related to device twin and module twin functionality
+"""
+
+
+class Twin(object):
+    """Represents a device twin or module twin
+
+    :ivar desired_properties: The desired properties for the Twin.  These are properties
+      which are sent _to_ the device or module to indicate the _desired_ state of the device
+      or module
+    :type desired_properties: dict, str, int, float, bool, or None (JSON compatible values)
+    :ivar reported_properties: The reported properties for the Twin.  These are properties
+      which are sent _from_ the device or module to indicate the _actual_ state of the device.
+    :type reported_properties: dict, str, int, float, bool, or None (JSON compatible values)
+    """
+
+    # OPEN QUESTION: can desired_properties and reported_properties only be dict, or can they be any JSON-compatible value?
```

Comment:
> OPEN QUESTION [](start = 5, length = 14)

is there a set of properties that remain constant for all devices and modules ?

Reply:
There is not.  the desired and reported items at the root are the only guaranteed items.

---
In reply to: [287529657](https://github.com/Azure/azure-iot-sdk-python-preview/pull/96#discussion_r287529657) [](ancestors = 287529657)